pkg/daemon/drain.go

```diff
@@ -217,33 +218,45 @@ func isSafeContainerRegistryConfChanges(oldIgnConfig, newIgnConfig ign3types.Con
 						constants.ContainerRegistryConfPath, regLoc, oldReg.Insecure, newReg.Insecure)
 					return false, nil
 				}
-				if oldReg.MirrorByDigestOnly == newReg.MirrorByDigestOnly {
-					// Ensure that all the old mirrors are present
-					for _, m := range oldReg.Mirrors {
-						if !searchRegistryMirror(m.Location, newReg.Mirrors) {
-							glog.Infof("%s: mirror %s has been removed in registry %s",
-								constants.ContainerRegistryConfPath, m.Location, regLoc)
+
+				// Ensure that all the old mirrors are present
+				for _, m := range oldReg.Mirrors {
+					if found, _ := searchRegistryMirror(m.Location, newReg.Mirrors); !found {
+						glog.Infof("%s: mirror %s has been removed in registry %s",
+							constants.ContainerRegistryConfPath, m.Location, regLoc)
+						return false, nil
+					}
+				}
+				// Ensure that any change to current does not unset pull-from-mirror="digest-only"
+				for _, m := range newReg.Mirrors {
+					if found, oldMirror := searchRegistryMirror(m.Location, oldReg.Mirrors); found {
+						if m.PullFromMirror != oldMirror.PullFromMirror && m.PullFromMirror != sysregistriesv2.MirrorByDigestOnly {
+
+							glog.Infof("%s: pull-from-mirror value for mirror %s has changed from %s to %s ",
+								constants.ContainerRegistryConfPath, m.Location, oldMirror.PullFromMirror, m.PullFromMirror)
 							return false, nil
 						}
 					}
-					// Ensure that any added mirror has mirror-by-digest-only set to true
-					for _, m := range newReg.Mirrors {
-						if !searchRegistryMirror(m.Location, oldReg.Mirrors) && !newReg.MirrorByDigestOnly {
-							glog.Infof("%s: mirror %s has been added in registry %s that has mirror-by-digest-only set to %t ",
-								constants.ContainerRegistryConfPath, m.Location, regLoc, newReg.MirrorByDigestOnly)
+
+				}
+
+				for _, m := range newReg.Mirrors {
```

Comment:
These two loops could be integrated … and the condition can probably be simplified. At the very least it would avoid two successive rounds of `searchRegistryMirror` for the same mirrors.